### PR TITLE
Modified the /:name/:slug endpoint to use handles

### PR DESCRIPTION
**Pull request by ryanpbrewster**
Now it is /:handle/:slug and it searches for a handle owner, not a user

@leogrim @andrewconner

### CODE REVIEW

File: kifi-backend/modules/shoebox/app/com/keepit/controllers/website/KifiSiteRouter.scala [hunk 1]
```diff
@@ -103,14 +104,18 @@ class KifiSiteRouter @Inject() (
...
-    lookupUsername(username) flatMap {
-      case (user, userRedirectStatusOpt) =>
-        libraryCommander.getLibraryBySlugOrAlias(user.id.get, LibrarySlug(slug)) map {
+  def serveWebAppIfLibraryFound(handle: Handle, slug: String) = WebAppPage { implicit request =>
+    lookupHandle(handle) flatMap {
+      case (handleOwner, userRedirectStatusOpt) =>
+        val libraryOpt = handleOwner match {
+          case Left(org) => libraryCommander.getLibraryBySlugOrAlias(org.id.get, LibrarySlug(slug))
+          case Right(user) => libraryCommander.getLibraryBySlugOrAlias(user.id.get, LibrarySlug(slug))
+        }
```
Comment (leogrim) [inline kifi-backend/modules/shoebox/app/com/keepit/controllers/website/KifiSiteRouter.scala:143]: Shouldn't `LibraryCommander.getLibraryBySlugOrAlias` take a space as well?

File: kifi-backend/modules/shoebox/app/com/keepit/controllers/website/KifiSiteRouter.scala [hunk 2]
```diff
@@ -119,6 +124,21 @@ class KifiSiteRouter @Inject() (
...
 
+  private def lookupHandle(handle: Handle): Option[(Either[Organization, User], Option[Int])] = {
+    val handleOwnerOpt = db.readOnlyMaster { implicit session => handleCommander.getByHandle(handle) }
+    handleOwnerOpt map {
+      case (handleOwner, isPrimary) =>
+        val foundHandle = handleOwner match {
+          case Left(org) => Handle.fromOrganizationHandle(org.getHandle)
+          case Right(user) => Handle.fromUsername(user.username)
+        }
+        if (foundHandle != handle) { // user moved or username normalization
```
Comment (leogrim) [inline kifi-backend/modules/shoebox/app/com/keepit/controllers/website/KifiSiteRouter.scala]: `// user moved or username normalization` => `// owner moved or handle normalization`

File: kifi-backend/modules/shoebox/app/com/keepit/controllers/website/KifiSiteRouter.scala (hunk 1 above)
Comment (leogrim) [inline kifi-backend/modules/shoebox/app/com/keepit/controllers/website/KifiSiteRouter.scala:143]: Ahah, actually you did change it, implicit conversion at work here.

Comment (ryanpbrewster) [inline kifi-backend/modules/shoebox/app/com/keepit/controllers/website/KifiSiteRouter.scala:143]: Gotta love those implicits...

File: kifi-backend/modules/shoebox/app/com/keepit/controllers/website/KifiSiteRouter.scala (hunk 2 above)
Comment (ryanpbrewster) [inline kifi-backend/modules/shoebox/app/com/keepit/controllers/website/KifiSiteRouter.scala]: :+1:

**Comment by andrewconner** [comment]
@ryanpbrewster We'd like to roll this live. Can you make it so that orgs _only_ route if the user is in an org? Otherwise, NotFound

**Comment by ryanpbrewster** [comment]
You want that, and not the Org experiment?

**Comment by andrewconner** [comment]
To bring meatspace conversations here, route if user is in the org experiment. NotFound otherwise.

**Comment by ryanpbrewster** [comment]
@andrewconner I think this is ready to go.

**Pull request merged**